src/sagemaker/session.py

```diff
@@ -314,11 +331,14 @@ def default_bucket(self):
         if self._default_bucket:
             return self._default_bucket
 
+        default_bucket = self._desired_default_bucket_name
         region = self.boto_session.region_name
-        account = self.boto_session.client(
-            "sts", region_name=region, endpoint_url=sts_regional_endpoint(region)
-        ).get_caller_identity()["Account"]
-        default_bucket = "sagemaker-{}-{}".format(region, account)
+
+        if not default_bucket:
+            account = self.boto_session.client(
+                "sts", region_name=region, endpoint_url=sts_regional_endpoint(region)
+            ).get_caller_identity()["Account"]
+            default_bucket = "sagemaker-{}-{}".format(region, account)
```

Comment:
nit: I'd move l. 334 to be right before the `if` statement because it's clearer that's it's basically just an `if`/`else` there. Or just make it an `if`/`else`:

```python
if self._desired_default_bucket_name
    default_bucket = self._desired_default_bucket_name
else
    ...
```

Reply:
I moved the line closer, but I like to avoid ifelses when possible.
I think it's more readable to have a default case get overridden under specific circumstances. Also guarantees that the variable always gets set.